File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        n = 1
        zeroFlag = False

        for x in nums:
            if x == 0 and zeroFlag == False:
                zeroFlag = True
                continue
            n *= x
        
        arr = []

        if zeroFlag:
            for x in nums:
                if x == 0:
                    arr.append(n)
                else:
                    arr.append(0)
        else:
            for x in nums:
                arr.append(int(n / x))

        return arr

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> freqTable;

        for (int i = 0; i < nums.size(); i++) {
            if (freqTable[nums[i]] >= 1) {
                return true;
            }
            freqTable[nums[i]]++;
        }

        return false;
    }
};
